src/main/java/com/auction/retailService/exception/ProductAlreadyExistException.java

```diff
@@ -0,0 +1,17 @@
+package com.auction.retailService.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import com.auction.retailService.constant.ErrorMessageConstant;
+
+@ResponseStatus (value = HttpStatus.PRECONDITION_FAILED , reason = ErrorMessageConstant.PRODUCT_ALREADY_EXIST)
+public class ProductAlreadyExistException extends RuntimeException{
+
+	private static final long serialVersionUID = 1L;
+	
+	public ProductAlreadyExistException(String prodName, String skuNumber, String comapnyName) {
+		super(ErrorMessageConstant.PRODUCT_ALREADY_EXIST+" productname: "+" sku Number: "+skuNumber+" companyname: "+comapnyName);
```

Comment:
Use StringBuilder or String.format. 

This type of code generates a lot of immutable strings that use more memory.  This is a better solution

```java 
// 1) Modify ErrorMessageConstant.PRODUCT_ALREADY_EXIST to contain %s variables 

//2) 
super(
    String.format(
        ErrorMessageConstant.PRODUCT_ALREADY_EXIST, prodName, skuNumber, companyName
    )
);
```

Also, get used to writing mnemonic variable names without typos. In Intellij every typo gets green underlines so you can fix it. 
https://docs.oracle.com/javase/tutorial/java/nutsandbolts/variables.html